tests/unit/test_variable_utilities.py

```diff
@@ -0,0 +1,163 @@
+        'string_variable_time_utc',
+        'group1/nested_string_variable_time_utc',
+        'non_string_variable',
+    }
+
+    requested_harmony_variables = [
+        HarmonyVariable({'fullPath': variable_path})
+        for variable_path in requested_variable_paths
+    ]
+
+    required_variables = {
+        '/string_variable_time_utc',
+        '/group1/nested_string_variable_time_utc',
+        '/non_string_variable',
+        '/coordinate1',  # Additional CF variable
+    }
+
+    mock_get_excluded_variables = mocker.patch(
+        'hoss.variable_utilities.get_excluded_variables',
+        return_value=set(
+            ['/string_variable_time_utc', '/group1/nested_string_variable_time_utc']
+        ),
+    )
+
+    expected_output = {'/non_string_variable', '/coordinate1'}
+
+    actual_output = get_processable_variables(
+        required_variables, requested_harmony_variables, mock_varinfo, logger
+    )
+
+    mock_get_excluded_variables.assert_called_once()
+    assert expected_output == actual_output
+
+
+def test_get_processable_variables_contains_no_string_variables(
+    mocker, mock_varinfo, logger
+):
+    """This test checks that the output string set matches the input string set
+    when no string variables are included in the request.
+
+    """
+    requested_variable_paths = {'non_string_variable'}
+
+    requested_harmony_variables = [
+        HarmonyVariable({'fullPath': variable_path})
+        for variable_path in requested_variable_paths
+    ]
+
+    required_variables = {
+        '/non_string_variable',
+        '/coordinate1',  # Additional CF variable
+    }
+
+    mock_get_excluded_variables = mocker.patch(
+        'hoss.variable_utilities.get_excluded_variables', return_value=set()
+    )
+
+    expected_output = {'/non_string_variable', '/coordinate1'}
+
+    actual_output = get_processable_variables(
+        required_variables, requested_harmony_variables, mock_varinfo, logger
+    )
+
+    mock_get_excluded_variables.assert_called_once()
+    assert expected_output == actual_output
+
+
+def test_get_processable_variables_exception(mocker, mock_varinfo, logger):
+    """This test checks that the OnlyInvalidVariablesRequested exception is thrown
+    when only string variables are requested.
+
+    """
+    requested_variable_paths = {
+        'string_variable_time_utc',
+        'group1/nested_string_variable_time_utc',
+    }
+
+    requested_harmony_variables = [
+        HarmonyVariable({'fullPath': variable_path})
+        for variable_path in requested_variable_paths
+    ]
+
+    required_variables = {
+        '/string_variable_time_utc',
+        '/group1/nested_string_variable_time_utc',
+        '/coordinate1',  # Additional CF variable
+    }
+
+    mock_get_excluded_variables = mocker.patch(
+        'hoss.variable_utilities.get_excluded_variables',
+        return_value=set(
+            ['/string_variable_time_utc', '/group1/nested_string_variable_time_utc']
```

Comment:
I'd update this to include a variable that is excluded but not in the requested variables. (right now it won't raise an exception)